File: src/main/java/net/hotamachisubaru/casino/Listener/RouletteGUIListener.java
package net.hotamachisubaru.casino.Listener;

import net.hotamachisubaru.casino.Casino;
import net.hotamachisubaru.casino.Manager.BetManager;
import net.hotamachisubaru.casino.Roulette.Roulette;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;

public class RouletteGUIListener implements Listener {

    net.hotamachisubaru.casino.Manager.BetManager betManager;


    @EventHandler
    public void onInventoryClick(InventoryClickEvent event) {
        if (event.getView().getTitle().equals("ルーレットホイール")) {
            event.setCancelled(true); // GUI操作をキャンセル
            handleRouletteAction(event);
        }
    }

    private void handleRouletteAction(InventoryClickEvent event) {
        if (event.getCurrentItem() == null) return;
        if (event.getCurrentItem().getItemMeta() == null) return;
        String bet = event.getCurrentItem().getItemMeta().getDisplayName();
        Player player = (Player) event.getWhoClicked();
        Casino plugin = Casino.getPlugin(Casino.class);

        // Validate and obtain bet amount, then execute roulette
        BetManager betManager = plugin.getBetManager();
        Integer betAmount = betManager.getBetAmount(player);
        if (betAmount == null) {
            player.sendMessage("賭け金が設定されていません。");
            return;
        }
        new Roulette(plugin).executeRoulette(player, bet, betAmount);

        // Clear bet amount and close inventory
        betManager.clearBet(player);
        player.closeInventory();
    }
}

File: src/main/java/net/hotamachisubaru/casino/Listener/RouletteGUIListener.kt
package net.hotamachisubaru.casino.Listener

import net.hotamachisubaru.casino.Casino
import net.hotamachisubaru.casino.Roulette.Roulette
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

class RouletteGUIListener : Listener {
    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        if (event.getView().getTitle() == "ルーレットホイール") {
            event.setCancelled(true) // GUI操作をキャンセル

            if (event.getCurrentItem() != null && event.getCurrentItem()!!.getItemMeta() != null) {
                val bet = event.getCurrentItem()!!.getItemMeta().getDisplayName()
                val player = event.getWhoClicked() as Player
                val plugin = Casino.getPlugin<Casino>(Casino::class.java)

                // 賭け金を取得してルーレットを実行
                val betAmount = plugin.getBetManager().getBet(player)
                Roulette().executeRoulette(player, bet, betAmount)

                // 賭け金をクリアしてインベントリを閉じる
                plugin.getBetManager().clearBet(player)
                player.closeInventory()
            }
        }
    }
}
